packages/engine-frontend/components/IntegrationSearch.tsx

```diff
@@ -36,83 +38,96 @@ export function IntegrationSearch({
     ccfg: connectorConfigs.find((ccfg) => ccfg.id === int.connector_config_id)!,
   }))
 
+  const categories = Array.from(
+    new Set(connectorConfigs.flatMap((ccfg) => ccfg.verticals)),
+  )
+
   const intsByCategory = ints?.reduce(
     (acc, int) => {
       int.ccfg.verticals.forEach((vertical) => {
-        acc[vertical] = (acc[vertical] || []).concat(int)
+        if (categoryFilter.length === 0 || categoryFilter.includes(vertical)) {
+          acc[vertical] = (acc[vertical] || []).concat(int)
+        }
       })
       return acc
     },
     {} as Record<string, typeof ints>,
   )
 
+  const onApplyFilter = (selected: string[]) => {
+    setCategoryFilter(selected)
+  }
+
   return (
     <div className={className}>
       {/* Search integrations */}
       <div className="mb-2 bg-background/95 pt-4 backdrop-blur supports-[backdrop-filter]:bg-background/60">
-        <form>
-          <div className="relative">
+        <div className="flex flex-row gap-2">
+          <div className="relative w-[450px]">
             {/* top-2.5 is not working for some reason due to tailwind setup */}
-            <Search className="absolute left-2 top-2 h-4 w-4 text-muted-foreground" />
+            <Search className="absolute left-2 top-2.5 h-4 w-4 text-muted-foreground" />
             <Input
               placeholder="Search"
               className="pl-8"
               value={searchText}
               onChange={(e) => setSearchText(e.target.value)}
             />
           </div>
-        </form>
+          <CheckboxFilter options={categories} onApply={onApplyFilter} />
```

Comment:
Can we show this only if number of categories > 1?

Reply:
Done in the last commit